compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11358,49 +11359,49 @@ private STNode parseTupleMemberTypeDescList() {
         }
 
         // Parse first typedesc, that has no leading comma
-        STNode typeDesc = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_TUPLE);
-
+        STNode typeDesc = parseMemberDescriptor();
         return parseTupleTypeMembers(typeDesc, typeDescList);
     }
 
-    private STNode parseTupleTypeMembers(STNode typeDesc, List<STNode> typeDescList) {
-        typeDesc = parseComplexTypeDescriptor(typeDesc, ParserRuleContext.TYPE_DESC_IN_TUPLE, false);
-
-        STNode tupleMemberRhs = parseTypeDescInTupleRhs();
-        if (tupleMemberRhs != null) {
-            typeDesc = STNodeFactory.createRestDescriptorNode(typeDesc, tupleMemberRhs);
-        }
-
+    private STNode parseTupleTypeMembers(STNode membertypeDesc, List<STNode> memberTypeDescList) {
+        STNode tupleMemberRhs;
         // Parse the remaining type descs
         while (!isEndOfTypeList(peek().kind)) {
-            if (typeDesc.kind == SyntaxKind.REST_TYPE) {
-                typeDesc = invalidateTypeDescAfterRestDesc(typeDesc);
+            if (membertypeDesc.kind == SyntaxKind.REST_TYPE) {
+                membertypeDesc = invalidateTypeDescAfterRestDesc(membertypeDesc);
                 break;
             }
 
             tupleMemberRhs = parseTupleMemberRhs();
             if (tupleMemberRhs == null) {
                 break;
             }
-
-            typeDescList.add(typeDesc);
-            typeDescList.add(tupleMemberRhs);
-            typeDesc = parseMemberDescriptor();
+            memberTypeDescList.add(membertypeDesc);
+            memberTypeDescList.add(tupleMemberRhs);
+            membertypeDesc = parseMemberDescriptor();
         }
 
-        typeDescList.add(typeDesc);
-        return STNodeFactory.createNodeList(typeDescList);
+        memberTypeDescList.add(membertypeDesc);
+        return STNodeFactory.createNodeList(memberTypeDescList);
     }
 
     private STNode parseMemberDescriptor() {
```

Comment:
Not from your change. Let's rename this to some like `parseMemberDescOrRestDescriptor()`or simply `parseTupleMember`.  Otherwise according to the spec, member descriptor means only `[annots] type-descriptor`.